internal/controller/phases.go

```diff
@@ -279,6 +282,41 @@ func (p *phaseReconciler) configmapRepository(ctx context.Context, labelSelector
 	return mr, nil
 }
 
+// getComponentsData returns components data based on if it's compressed or not.
+func getComponentsData(cm corev1.ConfigMap) (string, error) {
+	// Data is not compressed, return it immediately.
+	if cm.GetAnnotations()[compressedAnnotation] != "true" {
```

Comment:
We should also add docs about this annotation, If users compress manifests manually and try to use them in an air-gapped scenario, this annotation has to be set on the configmap.

Reply:
Hey! I added the documentation, that describes this process for air-gapped environments.